utils/misc/dbutils_test.go

```diff
@@ -128,3 +130,30 @@ func TestIdleTxTimeout(t *testing.T) {
 		require.NoError(t, tx.Commit())
 	})
 }
+
+func TestCommonPool(t *testing.T) {
+	pool, err := dockertest.NewPool("")
+	require.NoError(t, err)
+	postgresContainer, err := postgres.Setup(pool, t)
+	require.NoError(t, err)
+
+	conf := config.New()
+	conf.Set("DB.host", postgresContainer.Host)
+	conf.Set("DB.user", postgresContainer.User)
+	conf.Set("DB.name", postgresContainer.Database)
+	conf.Set("DB.port", postgresContainer.Port)
+	conf.Set("DB.password", postgresContainer.Password)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	db, err := misc.NewDatabaseConnectionPool(ctx, conf, stats.NOP, "test")
+	require.NoError(t, err)
+	require.NoError(t, db.Ping())
+	defer db.Close()
+	require.Equal(t, 40, db.Stats().MaxOpenConnections)
+
+	conf.Set("db.test.pool.maxOpenConnections", 5)
+	time.Sleep(10 * time.Second)
+	require.Equal(t, 5, db.Stats().MaxOpenConnections)
+	t.Log()
```

Comment:
10s seems to be a lot for this test. Can we make the config configurable and set some lower value to test it out.
```suggestion
	pool, err := dockertest.NewPool("")
	require.NoError(t, err)
	postgresContainer, err := postgres.Setup(pool, t)
	require.NoError(t, err)

	conf := config.New()
	conf.Set("DB.host", postgresContainer.Host)
	conf.Set("DB.user", postgresContainer.User)
	conf.Set("DB.name", postgresContainer.Database)
	conf.Set("DB.port", postgresContainer.Port)
	conf.Set("DB.password", postgresContainer.Password)
	conf.Set("db.pool.configUpdateInterval", "10ms")

	ctx, cancel := context.WithCancel(context.Background())
	defer cancel()
	db, err := misc.NewDatabaseConnectionPool(ctx, conf, stats.NOP, "test")
	require.NoError(t, err)
	require.NoError(t, db.Ping())
	defer func() {
		_ = db.Close()
	}()
	require.Equal(t, 40, db.Stats().MaxOpenConnections)

	conf.Set("db.test.pool.maxOpenConnections", 5)
	time.Sleep(100 * time.Millisecond)
	require.Equal(t, 5, db.Stats().MaxOpenConnections)
```